lib/rules/no-observers.js

```diff
@@ -34,13 +35,36 @@ module.exports = {
         if (ember.isModule(node, 'observer')) {
           report(node);
         }
+
+        const { callee } = node;
+
+        if (
+          (types.isThisExpression(callee.object) || types.isMemberExpression(callee)) &&
+          types.isIdentifier(callee.property) &&
+          callee.property.name === 'addObserver'
+        ) {
+          report(node);
+        }
       },
 
       Decorator(node) {
         if (ember.isObserverDecorator(node)) {
           report(node);
         }
       },
+
+      ImportDeclaration(node) {
+        const importSource = node.source.value;
+
+        if (
+          (importSource === '@ember/object' &&
+            node.specifiers.some(s => s.imported.name === 'observer')) ||
+          (importSource === '@ember/object/observers' &&
```

Comment:
Should we just disallow any import from `@ember/object/observers` (instead of just `addObserver`)?